src/utils/command.rs

```diff
@@ -46,24 +56,57 @@ where
     A: Display,
     T: Write,
 {
-    /// Queue the given command for later execution.
+    /// Queues the given command for further execution.
     ///
     /// Queued commands will be executed in the following cases:
-    /// - When you manually call `flush` on the given writer.
-    /// - When the buffer is to full, then the terminal will flush for you.
-    /// - Incase of `stdout` each line, because `stdout` is line buffered.
     ///
-    /// # Parameters
+    /// * When `flush` is called manually on the given type implementing `io::Write`.
+    /// * The terminal will `flush` automatically if the buffer is full.
+    /// * Each line is flushed in case of `stdout`, because it is line buffered.
+    ///
+    /// # Arguments
+    ///
     /// - [Command](./trait.Command.html)
     ///
     ///     The command that you want to queue for later execution.
     ///
-    /// # Remarks
-    /// - In the case of UNIX and windows 10, ANSI codes are written to the given 'writer'.
-    /// - In case of Windows versions lower than 10, a direct WinApi call will be made.
-    /// This is happening because windows versions lower then 10 do not support ANSI codes, and thus they can't be written to the given buffer.
-    /// Because of that there is no difference between `execute` and `queue` for those windows versions.
-    /// - Queuing might sound that there is some scheduling going on, however, this means that we write to the stdout without flushing which will cause commands to be stored in the buffer without them being written to the terminal.
+    /// # Examples
+    ///
+    /// ```rust
+    /// use std::io::{Write, stdout};
+    /// use crossterm::{Result, QueueableCommand, style::Print};
+    ///
+    ///  fn main() -> Result<()> {
+    ///     let mut stdout = stdout();
+    ///
+    ///     // `Print` will executed executed when `flush` is called.
+    ///     stdout
+    ///         .queue(Print("foo 1\n".to_string()))?
+    ///         .queue(Print("foo 2".to_string()))?;
+    ///
+    ///     // some other code (no execution happening here) ...
+    ///
+    ///     // when calling `flush` on `stdout`, all commands will be written to the stdout and therefore executed.
```

Comment:
Wrap this long line please otherwise the generated docs example will contain scrollbar.